patterns/vpc-lattice/main.tf

```diff
@@ -0,0 +1,377 @@
+    }, ]
+
+  }
+
+  tags = local.tags
+}
+
+resource "null_resource" "wait_for_crd_registration" {
+  triggers = {
+    build_number = "$(timestamp())"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOF
+      echo "Waiting for crd registration of aws api gateway controller..."
+      sleep 30
+    EOF
+  }
+
+  depends_on = [module.addons]
+}
+
+data "aws_iam_policy_document" "gateway_api_controller" {
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "vpc-lattice:*",
+      "iam:CreateServiceLinkedRole",
+      "ec2:DescribeVpcs",
+      "ec2:DescribeSubnets",
+      "ec2:DescribeTags",
+      "ec2:DescribeSecurityGroups",
+      "logs:CreateLogDelivery",
+      "logs:GetLogDelivery",
+      "logs:UpdateLogDelivery",
+      "logs:DeleteLogDelivery",
+      "logs:ListLogDeliveries",
+      "tag:GetResources",
+    ]
+  }
+}
+
+################################################################################
+# Demo applications
+################################################################################
+
+resource "helm_release" "demo_application" {
+  name             = "demo-application"
+  chart            = "./charts/demo-application"
+  create_namespace = true
+  namespace        = "apps"
+
+  depends_on = [null_resource.wait_for_crd_registration]
+}
+
+################################################################################
+# Update cluster security group to allow access from VPC Lattice
+################################################################################
+
+data "aws_ec2_managed_prefix_list" "vpc_lattice_ipv4" {
+  name = "com.amazonaws.${local.region}.vpc-lattice"
+}
+
+resource "aws_vpc_security_group_ingress_rule" "cluster_sg_ingress" {
+  security_group_id = module.eks.node_security_group_id
+
+  prefix_list_id = data.aws_ec2_managed_prefix_list.vpc_lattice_ipv4.id
+  ip_protocol    = "-1"
+}
+
+################################################################################
+# VPC Lattice service network
+################################################################################
+
+resource "aws_vpclattice_service_network" "this" {
+  name      = "my-services"
+  auth_type = "NONE"
+
+  tags = local.tags
+}
+
+resource "aws_vpclattice_service_network_vpc_association" "cluster_vpc" {
+  vpc_identifier             = module.cluster_vpc.vpc_id
+  service_network_identifier = aws_vpclattice_service_network.this.id
+}
+
+resource "aws_vpclattice_service_network_vpc_association" "client_vpc" {
+  vpc_identifier             = module.client_vpc.vpc_id
+  service_network_identifier = aws_vpclattice_service_network.this.id
+}
+
+resource "null_resource" "wait_for_lattice_resources" {
+  triggers = {
+    build_number = "$(timestamp())"
+  }
+  provisioner "local-exec" {
+    command = <<EOF
+      echo "Waiting for vpc lattice resources..."
+      sleep 120
+    EOF
+  }
+
+  depends_on = [helm_release.demo_application]
+}
```

Comment:
Have you tested with `time_sleep` other than `null_resource`?
https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep

Reply:
Good point. I've replaced both null_resource with time_sleep

Reply:
Awesome!